LibArchive+Utilities: Port ZipOutputStream and zip(1) to `Core::Stream`

krkk opened the pull request.

alimpfard commented on Userland/Libraries/LibArchive/Zip.cpp:
```diff
@@ -141,10 +141,11 @@ void ZipOutputStream::add_member(ZipMember const& member)
         .compressed_data = member.compressed_data.data(),
     };
-    local_file_header.write(m_stream);
+    TRY(local_file_header.write(*m_stream));
+    return {};
```
```suggestion
    return local_file_header.write(*m_stream);
```

alimpfard commented on Userland/Libraries/LibArchive/Zip.cpp:
```diff
@@ -189,7 +190,8 @@ void ZipOutputStream::finish()
         .comment = nullptr,
     };
-    end_of_central_directory.write(m_stream);
+    TRY(end_of_central_directory.write(*m_stream));
+    return {};
```
same?

alimpfard commented on Userland/Utilities/zip.cpp:
```diff
@@ -76,10 +73,11 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         member.crc32 = checksum.digest();
         member.is_directory = false;
-        zip_stream.add_member(member);
+        TRY(zip_stream.add_member(member));
+        return {};
```
And dis.

alimpfard reviewed (commented):
LGTM, but have some minor comments you can ignore if you want:

krkk commented:
note to the future self: commit category is wrong (LibCompress → LibArchive)

krkk commented:
changes since last push:
- applied suggested changes
- fixed commit category
- removed now unused `OutputStream& operator<<` helper function for ZipCompressionMethods
